### PR TITLE
Deploy App - E2E test

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This PR adds an E2E test where [Google MicroserviceDemo application](https://github.com/liqotech/microservices-demo) is deployed.

### Features
- [x] Deploy application from manifest
  * if some backend service fails, also the frontend sends an HTTP error
- [x] Order test
- [x] Enable Liqo (offload pods in different clusters)

# How Has This Been Tested?

- [x] In 2 local kind clusters
- [x] In github actions e2e kind clusters

### CODE REVIEW

**Comment by palexster** [comment]
Running tests, I have this strange crash of virtual-kubelets:
```root@liqo1-control-plane:/# kubectl logs -n liqo virtual-kubelet-eb425d5a-1886-43c8-a0e8-caaa851a7c28-859cb7zrj7
The connection to the server localhost:8080 was refused - did you specify the right host or port?
root@liqo1-control-plane:/# export KUBECONFIG=/etc/kubernetes/admin.conf 
root@liqo1-control-plane:/# kubectl logs -n liqo virtual-kubelet-eb425d5a-1886-43c8-a0e8-caaa851a7c28-859cb7zrj7
I1003 19:29:07.831227       1 namespaceMapper.go:69] namespaceNattingTable cache initialized
I1003 19:29:07.983833       1 root.go:265] setup ended
I1003 19:29:07.985089       1 podcontroller.go:261] Pod cache in-sync
I1003 19:29:07.985155       1 pod.go:201] receive GetPods
I1003 19:29:07.985264       1 pod.go:62] receive DeletePod "tester-remote-cl1"
E1003 19:29:07.985283       1 podcontroller.go:581] failed to delete pod "test-connectivity-cl1-91790a65-138d-41d2-8191-b0f07c878c19/tester-remote-cl1" in provider
I1003 19:29:07.985316       1 podcontroller.go:303] starting workers
I1003 19:29:07.985367       1 podcontroller.go:336] started workers
I1003 19:29:08.018197       1 virtualNodeupdater.go:159] correctly set pod CIDR from tunnel endpoint
I1003 19:29:08.018235       1 virtualNodeupdater.go:99] advertisement advertisement-eb425d5a-1886-43c8-a0e8-caaa851a7c28 is going to be deleted... set node status not ready
I1003 19:29:08.028499       1 virtualNodeupdater.go:116] delete all offloaded resources and advertisement
panic: interface conversion: interface {} is *v1beta1.EndpointSlice, not *v1.ConfigMap

goroutine 70 [running]:
github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/reflectors/outgoing.(*EndpointSlicesReflector).CleanupNamespace(0xc00052c5a0, 0xc0001272a0, 0x15)
	/go/src/github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/reflectors/outgoing/endpointSlices.go:222 +0x41c
github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/controller.(*OutgoingReflectorsController).startNamespaceReflection.func1(0xc00000e048, 0xc0001272a0, 0x15)
	/go/src/github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/controller/outgoingReflectorsController.go:126 +0x1e2
created by github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/controller.(*OutgoingReflectorsController).startNamespaceReflection
	/go/src/github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/controller/outgoingReflectorsController.go:121 +0x7aa

**Review by palexster — APPROVED** [review]

File: test/e2e/deploy_app_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"os"
+	"testing"
+	"time"
+
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+)
+
+const (
+	retries             = 36
```
Comment (mLavacca) [inline test/e2e/deploy_app_test.go:18]: Is there any reason for having 36 retries?

**Review by mLavacca — APPROVED** [review]

**Comment by mLavacca** [comment]
> Running tests, I have this strange crash of virtual-kubelets:
> 
> ```
> The connection to the server localhost:8080 was refused - did you specify the right host or port?
> root@liqo1-control-plane:/# export KUBECONFIG=/etc/kubernetes/admin.conf 
> root@liqo1-control-plane:/# kubectl logs -n liqo virtual-kubelet-eb425d5a-1886-43c8-a0e8-caaa851a7c28-859cb7zrj7
> I1003 19:29:07.831227       1 namespaceMapper.go:69] namespaceNattingTable cache initialized
> I1003 19:29:07.983833       1 root.go:265] setup ended
> I1003 19:29:07.985089       1 podcontroller.go:261] Pod cache in-sync
> I1003 19:29:07.985155       1 pod.go:201] receive GetPods
> I1003 19:29:07.985264       1 pod.go:62] receive DeletePod "tester-remote-cl1"
> E1003 19:29:07.985283       1 podcontroller.go:581] failed to delete pod "test-connectivity-cl1-91790a65-138d-41d2-8191-b0f07c878c19/tester-remote-cl1" in provider
> I1003 19:29:07.985316       1 podcontroller.go:303] starting workers
> I1003 19:29:07.985367       1 podcontroller.go:336] started workers
> I1003 19:29:08.018197       1 virtualNodeupdater.go:159] correctly set pod CIDR from tunnel endpoint
> I1003 19:29:08.018235       1 virtualNodeupdater.go:99] advertisement advertisement-eb425d5a-1886-43c8-a0e8-caaa851a7c28 is going to be deleted... set node status not ready
> I1003 19:29:08.028499       1 virtualNodeupdater.go:116] delete all offloaded resources and advertisement
> panic: interface conversion: interface {} is *v1beta1.EndpointSlice, not *v1.ConfigMap
> 
> goroutine 70 [running]:
> github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/reflectors/outgoing.(*EndpointSlicesReflector).CleanupNamespace(0xc00052c5a0, 0xc0001272a0, 0x15)
> 	/go/src/github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/reflectors/outgoing/endpointSlices.go:222 +0x41c
> github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/controller.(*OutgoingReflectorsController).startNamespaceReflection.func1(0xc00000e048, 0xc0001272a0, 0x15)
> 	/go/src/github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/controller/outgoingReflectorsController.go:126 +0x1e2
> created by github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/controller.(*OutgoingReflectorsController).startNamespaceReflection
> 	/go/src/github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/controller/outgoingReflectorsController.go:121 +0x7aa
> ```

This problem has been solved by #317

**Pull request merged**